userspace/libsinsp/event.cpp

```diff
@@ -1556,15 +1551,17 @@ const char* sinsp_evt::get_param_as_str(uint32_t id, OUT const char** resolved_s
 	//
 	if(param_info->type == PT_DYN && param_info->info != NULL)
 	{
-		uint8_t dyn_idx = *(uint8_t*)payload;
+		uint8_t dyn_idx = 0;
+		memcpy(&dyn_idx, param->m_val, sizeof(uint8_t));
 
 		if(dyn_idx < param_info->ninfo) {
 			const struct ppm_param_info* dyn_params =
 				(const struct ppm_param_info*)param_info->info;
 
-			payload += sizeof(uint8_t);
-			payload_len -= sizeof(uint8_t);
+			dyn_param = std::make_unique<sinsp_evt_param>(param->m_evt, param->m_idx,
+				param->m_val + sizeof(uint8_t), param->m_len - sizeof(uint8_t));
 
+			param = dyn_param.get();
```

Comment:
Could this be a stack allocation? e.g. an std::optional<sinsp_evt_param>

Reply:
Great idea! I forgot about `std::optional`

Reply:
done